### PR TITLE
Cloudsdk 123

### PR DESCRIPTION
**Pull request by szll**

### CODE REVIEW

File: lib/actions/ExtensionAction.js [hunk 1]
```diff
@@ -317,7 +320,12 @@ class ExtensionAction {
       if (!extensionInfo) {
         return logger.warn(`There is no extension in folder: './extensions/${extensionFolderName}'. Skipping... Please make sure that you only pass the folder name of the extension as argument.`)
       }
-      this.settings.attachExtension(extensionFolderName, extensionInfo, force)
+      this.settings.attachExtension(extensionFolderName, {id: extensionInfo.id, trusted: extensionInfo.trusted}, force)
+
+      // Create components.json if needed
+      if (extensionInfo.components && extensionInfo.components.length > 0) {
+        utils.generateComponentsJson(this.settings.getApplicationFolder(), extensionInfo.id, extensionInfo.components)
```
Comment (Rekhyt) [inline lib/actions/ExtensionAction.js]: I may have got some kind of missing link here - utils.generateComponents() expects one parameter "appSettings", supposedly an AppSettings object. This won't work, will it?

I also think we're missing a test case here that makes sure the method is called/not called under given conditions.

File: lib/utils/utils.js [hunk 2]
```diff
@@ -132,12 +134,53 @@ function getApplicationFolder () {
   return process.cwd()
 }
 
+function generateComponentsJson (appSettings) {
```
Comment (Rekhyt) [inline lib/utils/utils.js]: Missing JSDoc

File: test/utils/utils-test.js [hunk 3]
```diff
@@ -68,4 +78,91 @@ describe('utils', () => {
...
+      await fsEx.remove(projectDir)
+    })
+
+    it('should create components json', async () => {
+      utils.generateComponentsJson(appSettings)
+
+      const t1 = await fsEx.readJson(path.join(projectDir, THEMES_FOLDER, 'gmd', 'config', 'components.json'))
+      const t2 = await fsEx.readJson(path.join(projectDir, THEMES_FOLDER, 'ios', 'config', 'components.json'))
+
+      const result = {
```
Comment (Rekhyt) [inline test/utils/utils-test.js]: Would call this const "expected" or "expectedResult" maybe.

File: lib/actions/FrontendAction.js [hunk 4]
```diff
@@ -100,6 +104,7 @@ class FrontendAction {
     switch (action) {
       default:
       case 'start': {
+        utils.generateComponentsJson(this.appSettings)
```
Comment (Rekhyt) [inline lib/actions/FrontendAction.js]: I think we should have a test that makes sure the components.json is being generated upon frontend start.

File: test/utils/utils-test.js (hunk 3 above)
Comment (szll) [reply test/utils/utils-test.js]: done

File: lib/actions/ExtensionAction.js (hunk 1 above)
Comment (szll) [reply lib/actions/ExtensionAction.js]: missed that one ... done

File: lib/actions/FrontendAction.js [hunk 5]
```diff
@@ -112,7 +117,10 @@ class FrontendAction {
     }
 
     process.on('SIGINT', async () => {
-      await utils.deleteProcessFile('frontend', this.settingsFolder)
+      await Promise.all([
```
[the remark this answers is not in the document]
Comment (szll) [reply lib/actions/FrontendAction.js]: done

**Pull request merged**